src/Features/Core/Portable/Completion/CompletionOptionsMetadata.cs

```diff
@@ -0,0 +1,109 @@
+              EnterKeyBehavior: options.GetOption(EnterKeyBehavior, language),
+              SnippetsBehavior: options.GetOption(SnippetsBehavior, language),
+              HideAdvancedMembers: options.GetOption(HideAdvancedMembers, language),
+              ShowNameSuggestions: options.GetOption(ShowNameSuggestions, language),
+              ShowItemsFromUnimportedNamespaces: options.GetOption(ShowItemsFromUnimportedNamespaces, language),
+              UnnamedSymbolCompletionDisabled: options.GetOption(UnnamedSymbolCompletionDisabledFeatureFlag),
+              TargetTypedCompletionFilter: options.GetOption(TargetTypedCompletionFilterFeatureFlag),
+              TypeImportCompletion: options.GetOption(TypeImportCompletionFeatureFlag),
+              ProvideDateAndTimeCompletions: options.GetOption(ProvideDateAndTimeCompletions, language),
+              ProvideRegexCompletions: options.GetOption(ProvideRegexCompletions, language),
+              TimeoutInMillisecondsForExtensionMethodImportCompletion: options.GetOption(TimeoutInMillisecondsForExtensionMethodImportCompletion));
+
+        // feature flags
+
+        public static readonly Option2<bool> TypeImportCompletionFeatureFlag = new(nameof(CompletionOptions), nameof(TypeImportCompletionFeatureFlag),
+            CompletionOptions.Default.TypeImportCompletion,
+            new FeatureFlagStorageLocation("Roslyn.TypeImportCompletion"));
+
+        public static readonly Option2<bool> TargetTypedCompletionFilterFeatureFlag = new(nameof(CompletionOptions), nameof(TargetTypedCompletionFilterFeatureFlag),
+            CompletionOptions.Default.TargetTypedCompletionFilter,
+            new FeatureFlagStorageLocation("Roslyn.TargetTypedCompletionFilter"));
+
+        public static readonly Option2<bool> UnnamedSymbolCompletionDisabledFeatureFlag = new(nameof(CompletionOptions), nameof(UnnamedSymbolCompletionDisabledFeatureFlag),
+            CompletionOptions.Default.UnnamedSymbolCompletionDisabled,
+            new FeatureFlagStorageLocation("Roslyn.UnnamedSymbolCompletionDisabled"));
+
+        // This is serialized by the Visual Studio-specific LanguageSettingsPersister
+        public static readonly PerLanguageOption2<bool> HideAdvancedMembers = new(nameof(CompletionOptions), nameof(HideAdvancedMembers), CompletionOptions.Default.HideAdvancedMembers);
+
+        // This is serialized by the Visual Studio-specific LanguageSettingsPersister
+        public static readonly PerLanguageOption2<bool> TriggerOnTyping = new(nameof(CompletionOptions), nameof(TriggerOnTyping), CompletionOptions.Default.TriggerOnTyping);
+
+        public static readonly PerLanguageOption2<bool> TriggerOnTypingLetters = new(nameof(CompletionOptions), nameof(TriggerOnTypingLetters), CompletionOptions.Default.TriggerOnTypingLetters,
+            storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.TriggerOnTypingLetters"));
```

Comment:
aside... do we eventually want to get to he point that htings like RoamingProfileStorageLocation are moved up as well?  it seems like something odd to have at the workspace level.

Reply:
Definitely

Reply:
Although we might get forced to keep it in Workspaces if any public options use it.